### DISCUSSION TITLE
Usernames in Postal

### DISCUSSION THREAD

**Discussion by mryanbc** [postalserver/postal#3079] [Help with using Postal]
Is it possible to have Postal use the @ in the usernames for outgoing mail? As some third party systems expect the username to have an @domain.com.

**Comment by willpower232** [comment]
If you're using `AUTH PLAIN` or `AUTH LOGIN` then the username is discarded. If you're using CRAM MD5 then it must be in the `organisation/server` format.

**Reply by mryanbc** [reply]
Is there a way to set it to AUTH PLAIN or AUTH LOGIN in the Postal Config or anywhere else?

**Reply by willpower232** [reply]
If you have a say in setting it, then it will be in the client that connects to Postal. Postal doesn't care which you use.

**Reply by mryanbc** [reply]
I see, so its the client. No then... As a third party app works with mailgun but not postal. tried on 465 and 587, I assume its the username issue being organisation/server, as it is an old app. No access to error logs from client side.

**Reply by willpower232** [reply]
You can also watch the logs on the server to see what actually happens from the client to confirm what username it is presenting the server.

**Reply by mryanbc** [reply]
Where are the logs?

**Reply by willpower232** [reply]
If you're using the default installation, you can do `postal logs smtp`

https://github.com/postalserver/install/blob/c82658da7fa16615a2fba9bae2d02c860816fdb3/bin/postal#L184-L186

**Reply by mryanbc** [reply]
Did that, not seeing anything related. I think the third party app is not connecting on its own. This is an old Payment application that sends emails out as receipts. The payment still works but the emails are funny, works with some smtp providers and not others...

**Reply by mryanbc** [reply]
postal-smtp-1  | 2024-08-06 20:37:32 +0000 DEBUG  Client identified as (Hiding my IP) trace_id=WWIVFDTO compo                                                                                                                                nent=smtp-server
postal-smtp-1  | 2024-08-06 20:37:32 +0000 WARN   Detected line with invalid line ending (missing <CR>) trace                                                                                                                                _id=WWIVFDTO component=smtp-server
postal-smtp-1  | 2024-08-06 20:37:32 +0000 DEBUG  <= �k��ď��A#��WA��&Y���A�n�[O��� Bb�}͸��,��G����YR
                                                                                                    r�O��d��$̨̩                                                                                                                                �+�/�,�0�       �� trace_id=WWIVFDTO component=smtp-server
postal-smtp-1  | 2024-08-06 20:37:32 +0000 DEBUG  => 502 Invalid/unsupported command trace_id=WWIVFDTO compon                                                                                                                                ent=smtp-server
postal-smtp-1  | 2024-08-06 20:37:32 +0000 WARN   Detected line with invalid line ending (missing <CR>) trace                                                                                                                                _id=WWIVFDTO component=smtp-server
postal-smtp-1  | 2024-08-06 20:37:32 +0000 DEBUG  <= ���/5 trace_id=WWIVFDTO component=smtp-server
postal-smtp-1  | 2024-08-06 20:37:32 +0000 DEBUG  => 502 Invalid/unsupported command trace_id=WWIVFDTO compon                                                                                                                                ent=smtp-server
postal-smtp-1  | 2024-08-06 20:37:32 +0000 WARN   Detected line with invalid line ending (missing <CR>) trace                                                                                                                                _id=WWIVFDTO component=smtp-server
postal-smtp-1  | 2024-08-06 20:37:32 +0000 DEBUG  <= �mtp.blackcat.technology� trace_id=WWIVFDTO component=sm                                                                                                                                tp-server
postal-smtp-1  | 2024-08-06 20:37:32 +0000 DEBUG  => 502 Invalid/unsupported command trace_id=WWIVFDTO compon                                                                                                                                ent=smtp-server
postal-smtp-1  | 2024-08-06 20:37:32 +0000 DEBUG  Connection closed trace_id=WWIVFDTO component=smtp-server

**Reply by mryanbc** [reply]
postal-smtp-1  | 2024-08-06 20:42:51 +0000 DEBUG  Client identified as (Hiding my IP) trace_id=54AB9BT1 component=smtp-server
postal-smtp-1  | 2024-08-06 20:42:51 +0000 WARN   Detected line with invalid line ending (missing <CR>) trace_id=54AB9BT1 component=smtp-server
postal-smtp-1  | 2024-08-06 20:42:51 +0000 DEBUG  <= ~z
                                                       ��`�.*�Rzw��e�!�cK����NV'�
�       � trace_id=54AB9BT1 component=smtp-server
postal-smtp-1  | 2024-08-06 20:42:51 +0000 DEBUG  => 502 Invalid/unsupported command trace_id=54AB9BT1 component=smtp-server
postal-smtp-1  | 2024-08-06 20:42:51 +0000 WARN   Detected line with invalid line ending (missing <CR>) trace_id=54AB9BT1 component=smtp-server
postal-smtp-1  | 2024-08-06 20:42:51 +0000 DEBUG  <= ��/5E�mtp.blackcat.technology
                                                                                   trace_id=54AB9BT1 component=smtp-server
postal-smtp-1  | 2024-08-06 20:42:51 +0000 DEBUG  => 502 Invalid/unsupported command trace_id=54AB9BT1 component=smtp-server
postal-smtp-1  | 2024-08-06 20:42:51 +0000 DEBUG  Connection closed trace_id=54AB9BT1 component=smtp-server

**Reply by mryanbc** [reply]
This is the Postal Logs. Also I downloaded an SMTP test app that works BUT when i turn a setting  'Is secured connection' on then i get the same logs above as I do with the payment app. I am using port 587. TLS/SSL Certs are valid and working.

**Reply by willpower232** [reply]
I can only presume that using port 587 is making your client think the connection has implicit TLS whereas Postal only supports explicit TLS via STARTTLS. That would explain the garbled lines.

**Reply by mryanbc** [reply]
I am thinking the same. I can move it to 465 and see. The app works on 465 with mailgun, maybe that reveals more information.

**Reply by mryanbc** [reply]
on 465 it is giving me:
2024-08-09 09:48:54 +0000 DEBUG  Client identified as 195.158.70.216 trace_id=CYTCXVDM compo                                                                                                                                nent=smtp-server
postal-smtp-1  | 2024-08-09 09:48:54 +0000 WARN   Detected line with invalid line ending (missing <CR>) trace                                                                                                                                _id=CYTCXVDM component=smtp-server
postal-smtp-1  | 2024-08-09 09:48:54 +0000 DEBUG  <= ~z��q�#�Uʻ�Ա/h��okm��h(��A trace_id=CYTCXVDM component=s                                                                                                                                mtp-server
postal-smtp-1  | 2024-08-09 09:48:54 +0000 DEBUG  => 502 Invalid/unsupported command trace_id=CYTCXVDM compon                                                                                                                                ent=smtp-server
postal-smtp-1  | 2024-08-09 09:48:54 +0000 WARN   Detected line with invalid line ending (missing <CR>) trace                                                                                                                                _id=CYTCXVDM component=smtp-server
postal-smtp-1  | 2024-08-09 09:48:54 +0000 DEBUG  <= �
�       � trace_id=CYTCXVDM component=smtp-server
postal-smtp-1  | 2024-08-09 09:48:54 +0000 DEBUG  => 502 Invalid/unsupported command trace_id=CYTCXVDM compon                                                                                                                                ent=smtp-server
postal-smtp-1  | 2024-08-09 09:48:54 +0000 WARN   Detected line with invalid line ending (missing <CR>) trace                                                                                                                                _id=CYTCXVDM component=smtp-server
postal-smtp-1  | 2024-08-09 09:48:54 +0000 DEBUG  <= ��/5E�mtp.blackcat.technology
                                                                                   trace_id=CYTCXVDM componen                                                                                                                                t=smtp-server
postal-smtp-1  | 2024-08-09 09:48:54 +0000 DEBUG  => 502 Invalid/unsupported command trace_id=CYTCXVDM compon                                                                                                                                ent=smtp-server
postal-smtp-1  | 2024-08-09 09:48:54 +0000 DEBUG  Connection closed trace_id=CYTCXVDM component=smtp-server

**Reply by mryanbc** [reply]
@willpower232 Is there a way to force Postal to go implicit TLS?

**Reply by willpower232** [reply]
Unless you know enough ruby to figure it out, I don't think it is possible I'm afraid.

**Reply by mryanbc** [reply]
I see, so no yaml configuration to just change.

**Reply by syserr0r** [reply]
Thank-you so much @willpower232 - I couldn't understand why most of my SMTP clients could send using any username just fine but some required the username set in the `org/server` format.

There seemed to be conflicting information on this is several places (https://github.com/postalserver/postal/issues/1930 and the documentation in Postal for the 'Credentials' screen and 'Receiving E-mail')

In hindsight I can see it mentioned in https://docs.postalserver.io/features/smtp-authentication, but I skimmed over it and missed it. Is there any chance to make this more obvious in the UI itself?

**Reply by willpower232** [reply]
I can only merge PRs for the docs I'm afraid so if you make it different in the Postal UI, I don't know when that would be merged and shipped
